### ISSUE TITLE
Sporadic but reproducible build errors with Travis CI

### ISSUE THREAD

**Issue by bel2125** [travis-ci/travis-ci#6497]
I am the maintainer of the [CivetWeb](https://github.com/civetweb/civetweb) web server, written in C. It uses Travis CI to build a lot of different configurations (currently 199) - different configuration means different preprocessor defines of different operating systems (Linux, OSX). Sometimes building one of the configuration fails for unknown reasons, and succeeds if it is restarted.

In order to reproduce this more systematically, I created 10 commits with identical code (only a comment text file changed). The builds are
1. [716](https://travis-ci.org/civetweb/civetweb/builds/152590847) - all OK
2. [717](https://travis-ci.org/civetweb/civetweb/builds/152591137)
   - [717.149](https://travis-ci.org/civetweb/civetweb/jobs/152591300) error without any output
   - [717.172](https://travis-ci.org/civetweb/civetweb/jobs/152591323) error "No output has been received in the last 10m0s"
3. [718](https://travis-ci.org/civetweb/civetweb/builds/152591393)
   - [718.96](https://travis-ci.org/civetweb/civetweb/jobs/152591500) error "apt-get install failed"
   - [718.160](https://travis-ci.org/civetweb/civetweb/jobs/152591564) error without any output
   - [718.176](https://travis-ci.org/civetweb/civetweb/jobs/152591580) error without any output
   - [718.178](https://travis-ci.org/civetweb/civetweb/jobs/152591582) error without any output
   - [718.194](https://travis-ci.org/civetweb/civetweb/jobs/152591598) error without any output
4. [719](https://travis-ci.org/civetweb/civetweb/builds/152591637)
   - 156, 159 and 169 error without any output
5. [720](https://travis-ci.org/civetweb/civetweb/builds/152591928)
   - 140 fails without any output
   - 142 "No output has been received in the last 10m0s" (before the build)
   - 151 "No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself." (after the build but before the CI tests)
   - 168 "No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself." (in between the CI tests)
   - 177  "No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself." (after the CI tests)
   - 179 "No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself." (after the CI tests)
6. [721](https://travis-ci.org/civetweb/civetweb/builds/152592210) - all OK
7. [722](https://travis-ci.org/civetweb/civetweb/builds/152592516) - all OK
8. [723](https://travis-ci.org/civetweb/civetweb/builds/152592843) 
   - [723.199](https://travis-ci.org/civetweb/civetweb/jobs/152593052) failed without any output
9. [724](https://travis-ci.org/civetweb/civetweb/builds/152593096) (number 724 is used twice)
   - [724.19](https://travis-ci.org/civetweb/civetweb/jobs/152593127) establishing a HTTP connection during the test failed
   - 152, 153, 154 failed without output
   - 199 is not started yet while the next build is already running. The time is shown as a "-" only.
10. [724 again](https://travis-ci.org/civetweb/civetweb/builds/152593333) - yes, build number 724 is used twice
    - currently in progress, ... I took a screenshot of this moment, if it helps
    - build 1-161 completed without error
    - 162,163 did not start (shown in yellow as pending, time is shown as -)
    - 164,165,166 completed
    - 167 did not start (shown in yellow as pending, time is shown as -)
    - 168-172 completed
    - rest is not started yet (shown in yellow as pending, time is shown as -)

All 10 builds are absolutely identical, yet different configurations fail. Actually also [715](https://travis-ci.org/civetweb/civetweb/builds/152480555) is identical, [715.178](https://travis-ci.org/civetweb/civetweb/jobs/152480747) failed without output.

Is this some instability of the build VM?
Can I change something in the build configuration to avoid this?
Is there a way to automatically restart a build failed due to "No output has been received in the last 10m0s" and "no output at all"?

**Comment by bel2125** [comment]
This kind of failing build a sporadic, but actually often occurring scenario:
For the most recent build [728](https://travis-ci.org/civetweb/civetweb/builds/153999218), again several jobs failed without producing any line of output: [144](https://travis-ci.org/civetweb/civetweb/jobs/153999370), [145](https://travis-ci.org/civetweb/civetweb/jobs/153999371), [146](https://travis-ci.org/civetweb/civetweb/jobs/153999372) and [177](https://travis-ci.org/civetweb/civetweb/jobs/153999406). 

I am going to restarted these 4 builds.

It seems the "fails without any output" error only occurs for the OSX builds. Linux never had that in the 10 tests - Linux only had two "cannot communicate with HTTP"-kind of errors.

If there is anything else I can do to help analyzing this issue, let me know.

**Comment by bel2125** [comment]
It seems, I'm not the only one with these problems. Some hours after I reported this issue #6500 was submitted describing a situation similar to  ~50% of the failed builds.

Another ~40% are "No output has been received in the last 10m0s" - issue #6512 reported some hours ago seems to address the same problem.

Probably both are different symptoms of the same root cause "the build VM stalled".
In my "10x199 build jobs" test, I saw both symptoms only for OSX, not for Linux.

The remaining 10% are some communication problems - maybe problems with the network connection of the build environment. I can check if I find any similar issue.
